src/ngraph/op/slice_input.cpp

```diff
@@ -0,0 +1,134 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <vector>
+
+#include "ngraph/op/slice_input.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+const string op::SliceInput::type_name("SliceInput");
+
+op::SliceInput::SliceInput(const Output<Node>& value,
+                           std::ptrdiff_t axis,
+                           std::ptrdiff_t start,
+                           std::ptrdiff_t stride,
+                           std::ptrdiff_t part_size,
+                           std::ptrdiff_t end)
+    : Op({value})
+    , m_axis(axis)
+    , m_start(start)
+    , m_stride(stride)
+    , m_part_size(part_size)
+    , m_end(end)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::SliceInput::validate_and_infer_types()
+{
+    // For now, keep things simple; no padding, forwards stepping, etc.
+    NODE_VALIDATION_CHECK(this, 0 <= m_axis, "Negative axis is not supported.");
```

Comment:
Now we are not supporting m_axis, m_start, m_end and other parameters as a dynamic arguments for the Inference Engine TensorIterator.
Could we (together ngraph and dldt) think about how to make these parameters dynamic? It could add more flexibility in the future.